src/ui/public/flyout/flyout_session.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { EventEmitter } from 'events';
+import ReactDOM from 'react-dom';
+
+// TODO: Remove once EUI has typing for EuiFlyout and EuiFlyoutBody
+declare module '@elastic/eui' {
```

Comment:
__note:__ does this really give us any benefit comparing to just `ts-ignore` like we do for other cases?

We talked about that inline `declare module` thing with @timroes the other day and obviously the ideal option would be to have types file in "eui" package, even very simple and incomplete types would be better than that, __but__ only if eui owners would commit to keep these typings up to date. There is no need to write them all, just keep eye on existing types during review. Is it something we can do? Who is the owner of this package? @cjcenizal ?

Reply:
Soon as we have a new eui package I can get rid of this. I went into eui and added the typings (https://github.com/elastic/eui/pull/1001).  

I think @chandlerprall is taking charge of the typescript efforts in EUI.  I'm not sure what the plan is, but it seems a lot easier to upkeep if EUI used typescript instead of maintaining separate typings definitions. Perhaps that is the long term plan.

I'll wait for a new eui to submit this PR so I can get rid of this.

Reply:
Yep, Chandler owns the code side of EUI now. I agree that converting all of EUI to TS would make the most sense for ease-of-maintenance.

Reply:
Converting EUI to TS is the most likely solution. Either way we know we need to properly support TS consumers.